libindy/src/services/wallet/encryption.rs

```diff
@@ -7,51 +7,60 @@ use utils::crypto::chacha20poly1305_ietf::ChaCha20Poly1305IETF;
 
 use errors::common::CommonError;
 use errors::wallet::WalletError;
+use services::wallet::WalletRecord;
+
+use super::wallet::Keys;
+use super::storage::{Tag, TagName, StorageEntity};
+
+
+pub(super) fn encrypt_tag_names(tag_names: &[&str], tag_name_key: &[u8], tags_hmac_key: &[u8]) -> Vec<TagName> {
+    let mut encrypted_tag_names = Vec::new();
+
+    for tag_name in tag_names {
+        let tag_name_bytes = tag_name.as_bytes();
+        let tag_name = if tag_name_bytes.len() > 0 && tag_name_bytes[0] as char == '~' {
+            TagName::OfPlain(
+                encrypt_as_searchable(&tag_name_bytes[1..], tag_name_key, tags_hmac_key)
+            )
+        } else {
+            TagName::OfEncrypted(
+                encrypt_as_searchable(tag_name_bytes, tag_name_key, tags_hmac_key)
+            )
+        };
+        encrypted_tag_names.push(tag_name)
+    }
 
-use super::storage::{Tag, TagName};
-
-pub(super) fn encrypt_tag_names(tag_names: &str, tag_name_key: &[u8], tags_hmac_key: &[u8]) -> Result<Vec<TagName>, WalletError> {
-    let tag_names: Vec<String> = serde_json::from_str(tag_names)
-        .map_err(|err| CommonError::InvalidStructure(format!("Cannot deserialize tag_names")))?;
-
-    let etag_names = tag_names
-        .iter()
-        .map(|tag_name|
-            if tag_name.starts_with("~") {
-                TagName::OfPlain(encrypt_as_searchable(
-                    &tag_name.as_bytes()[1..], tag_name_key, tags_hmac_key))
-            } else {
-                TagName::OfEncrypted(encrypt_as_searchable(tag_name.as_bytes(), tag_name_key, tags_hmac_key))
-            })
-        .collect::<Vec<TagName>>();
-
-    Ok(etag_names)
+    encrypted_tag_names
 }
```

Comment:
Why did you revert these functions?

Reply:
Sorry, I didnt look at the content during the merge resolution. It slipped to me that you refactored them meanwhile. I thought it was outdated. I redid the merge, and your changes are active